psydac/api/tests/test_api_2d_compatible_spaces.py

```diff
@@ -86,95 +91,126 @@ def run_system_1_2d_dir(f0, sol, ncells, degree):
-    
-    phi1 = FemField(V1h)
-    phi2 = FemField(V2h)
+            x1, info = sp_minres(A1, b1, tol=tol)
+            print('Boundary solution with scipy.sparse: success = {}'.format(info == 0))
 
-    
-    phi1.coeffs[0][s11:e11+1, s12:e12+1] = x[:(e11+1)*(e12+1)].reshape((e11+1-s11, e12+1-s12))
-    phi1.coeffs[1][s21:e21+1, s22:e22+1] = x[(e11+1)*(e12+1):v2h_s].reshape((e21+1-s21, e22+1-s22))
-    phi2.coeffs[s31:e31+1, s32:e32+1]    = x[v2h_s:-1].reshape((e31-s31+1, e32-s32+1))
-    
-        # ... compute norms
-    l2_error = l2norm_F_h.assemble(F=phi1)
-    return l2_error
+            x0, info = sp_minres(A0, b0 - A0.dot(x1), tol=tol)
+            print('Interior solution with scipy.sparse: success = {}'.format(info == 0))
+
+            # Solution is sum of boundary and interior contributions
+            x = x0 + x1
+
+        else:
+            x, info = sp_minres(A0, b0, tol=tol)
+            print('Solution with scipy.sparse: success = {}'.format(info == 0))
+
+        # Convert to stencil format
+        x = array_to_stencil(x, Xh.vector_space)
+
+    else:
+
+        phi_h, info = equation_h.solve(info=True)
+        x = phi_h.coeffs
+        print(info)
+
+    # Numerical solution: velocity field
+    # TODO: allow this: uh = FemField(V1h, coeffs=x[0:2]) or similar
+    uh = FemField(V1h)
+    uh.coeffs[0][:] = x[0][:]
+    uh.coeffs[1][:] = x[1][:]
+
+    # Numerical solution: pressure field
+    # TODO: allow this: uh = FemField(V2h, coeffs=x[2])
+    ph = FemField(V2h)
+    ph.coeffs[:] = x[2][:]
+
+    # Compute norms of exact solution
+    x1, x2 = domain.coordinates
+    ue_1 = lambdify([x1, x2], ue[0])
+    ue_2 = lambdify([x1, x2], ue[1])
+    pe_c = lambdify([x1, x2], pe   )
+    l2_norm_ue = np.sqrt(V1h.spaces[0].integral(lambda *x: ue_1(*x)**2 + ue_2(*x)**2))
+    l2_norm_pe = np.sqrt(V2h.integral(lambda *x: pe_c(*x)**2))
+
+    # Average value of the pressure (should be 0)
+    domain_area = V2h.integral(lambda x1, x2: 1.0)
+    p_avg = V2h.integral(ph) / domain_area
+
+    # L2 error norm of the velocity field
+    error_u   = [ue[0]-u[0], ue[1]-u[1]]
+    l2norm_u  = Norm(error_u, domain, kind='l2')
+    l2norm_uh = discretize(l2norm_u, domain_h, V1h)
+
+    # L2 error norm of the pressure, after removing the average value from the field
+    l2norm_p  = Norm(pe - (p - p_avg), domain, kind='l2')
+    l2norm_ph = discretize(l2norm_p, domain_h, V2h)
+
+    # Compute error norms
+    l2_error_u = l2norm_uh.assemble(u = uh)
+    l2_error_p = l2norm_ph.assemble(p = ph)
+
+    print()
+    print('Relative l2_error(u) = {}'.format(l2_error_u / l2_norm_ue))
+    print('Relative l2_error(p) = {}'.format(l2_error_p / l2_norm_pe))
+    print('Average(p) = {}'.format(p_avg))
+
+    return locals()
 
 #==============================================================================
-def run_system_3_2d_dir(uex, f, alpha, ncells, degree):
+def run_maxwell_time_harmonic_2d_dir(uex, f, alpha, ncells, degree):
 
```

Comment:
same comment as above (cite reference if relevant), and also for other equations below

Reply:
@saidctb Is there a reference for this test case?

Reply:
no there is no reference